internal/controller/volumereplicationgroup_controller.go

```diff
@@ -763,6 +858,17 @@ func (v *VRGInstance) separatePVCsUsingStorageClassProvisioner(pvcList *corev1.P
 			}
 		}
 
+		if !replicationClassMatchFound {
```

Comment:
If CG is enabled, and SC has the replicationID label, then we should error out if we do not find a matching VGRClass, and not default the PVC to Volsync.

This either gets added here or with the PR #1487